src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -851,7 +851,14 @@ namespace winrt::TerminalApp::implementation
         });
         // Necessary for fly-out sub items to get focus on a tab before collapsing. Related to #15049
         newTabFlyout.Closing([this](auto&&, auto&&) {
-            _FocusCurrentTab(true);
+            if (const auto p = CommandPaletteElement(); p && p.Visibility() == Visibility::Visible)
```

Comment:
With this we now have 3 places where we ask for the visibility of the `CommandPaletteElement()`. I think this shows that it isn't just a pattern/occurrence. If you wanted to, we could now consider deduplicating them. One option to do that would be a function like this:

```cpp
bool TerminalPage::_commandPaletteIs(Visibility visibility)
{
    const auto p = CommandPaletteElement()
    return p && p.Visibility() == visibility;
}
```

Then we could write:
```cpp
if (!_commandPaletteIs(Visibility::Visible))
{
    _FocusCurrentTab(true);
}
```

But it's also very basic code so I don't mind not doing this at all.

Reply:
To build off Leonard's "we now have 3 places"...

we now have like a hundred places where we do manual visibility checks (search box, command palette, settings UI, etc.) and manual focus yeeting.

It makes me wonder... are we using the UI framework wrong?

Reply:
Indubitably. But the Terminal also does have a bit of a unique UX pattern, where we _almost always_ want the TermControl focused, kinda like a browser. I'm not sure how much other precedence there is for that in WinUI apps

Reply:
Meh. Fair. I'm more invested in Leonard's suggestion, so I may block for it.